source/modulo_components/src/ComponentInterface.cpp

```diff
@@ -39,8 +37,10 @@ ComponentInterface::ComponentInterface(
 
   this->add_predicate("in_error_state", false);
 
+  this->rate_ = this->get_parameter_value<double>("rate");
+  this->period_ = 1.0 / this->rate_;
```

Comment:
Do we really need to have a member variable for the period? As in, couldn't we just call the get_period() function which in C++ could be a constexpr `1 / rate_`. Clearly this is not a mistake, just trying to understand if it's for ease of use or serves a purpose.

Same goes for the python code.

Reply:
I wanted to save one operation...No there is no other purpose, I will remove it :+1:

Reply:
no strong opinion on this honestly. And I do hear the argumentation of saving one operation. The reason I asked is that we rarely use period in our derived classes.

Reply:
> The reason I asked is because we rarely use period in our derived classes.

We have several motion components that are interested in the component period for sinusoidal motion or to keep track of elapsed time. At the end, it's all about providing the helpers to simplify the work of the users.

Reply:
You are right yes. Then let's leave it and resolve this question. Thank you!